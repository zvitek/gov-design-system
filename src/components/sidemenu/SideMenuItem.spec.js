import { shallowMount } from '@vue/test-utils'
import GovSideMenuItem from '@components/sidemenu/SideMenuItem'

let wrapper

describe('GovSideMenuItem', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovSideMenuItem)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovSideMenuItem')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should have badge slot', () => {
        const slotBadge = '<div> Content </div>'
        wrapper = shallowMount(GovSideMenuItem, {
            slots: {
                badge: slotBadge
            }
        })
        expect(wrapper.find('.gov-sidenav__link :last-child').html()).toBe(slotBadge)
    })

    it('should have icon slot', () => {
        const slotIcon = '<div> Content </div>'
        wrapper = shallowMount(GovSideMenuItem, {
            slots: {
                icon: slotIcon
            }
        })
        expect(wrapper.find('.gov-sidenav__link :first-child').html()).toBe(slotIcon)
    })
})
