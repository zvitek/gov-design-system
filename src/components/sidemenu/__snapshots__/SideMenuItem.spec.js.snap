// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovSideMenuItem render correctly 1`] = `
<li class="gov-sidenav__item"><a class="gov-sidenav__link">
        <!----> </a></li>
`;
